src/components/GlobalAutoComplete.tsx

```diff
@@ -0,0 +1,60 @@
+import axios from "axios";
+import { useEffect, useState } from "react";
+import TextField from "@mui/material/TextField";
+import Autocomplete from "@mui/material/Autocomplete";
+
+const GlobalAutoComplete = (props: any) => {
+  const [data, setData] = useState([{ id: "", name: "" }]);
+  let requestTimeout: NodeJS.Timeout | null = null;
+
+  const getOptionsByProfix = async (
+    event: any,
+    value: string,
+    path: string
+  ) => {
+    console.log(new Date().toISOString(), value);
+    if (requestTimeout) {
+      clearTimeout(requestTimeout);
+    }
+    if (value.length > 2) {
+      requestTimeout = setTimeout(() => {
+        console.log(
+          "value:",
+          value,
+          "is send to server at:",
+          new Date().toISOString()
+        );
+        axios
```

Comment:
I recommend that you pass the fetch function in the props, such as a fetchAutoCompleteValues function that accepts the value as a parameter.

but what you did is ok too, I loved your component, simple and clear!